dewy/document/router.py

```diff
@@ -15,15 +28,16 @@
 
 
 async def ingest_document(
-    document_id: int, pg_pool: asyncpg.Pool, config: Config
+    document_id: int,
+    pg_pool: asyncpg.Pool,
+    config: Config,
+    request: Union[IngestContent, IngestURL],
 ) -> None:
     try:
-        url, embeddings = await CollectionEmbeddings.for_document_id(
-            pg_pool, config, document_id
-        )
-        if url.startswith("error://"):
-            raise RuntimeError(url.removeprefix("error://"))
-        await embeddings.ingest(document_id, url)
+        if isinstance(request, IngestURL) and request.url.startswith("error://"):
+            raise RuntimeError(request.url.removeprefix("error://"))
```

Comment:
Is this for testing?

Reply:
Yeah. This was added for the unit test that verifies errors from ingestion are properly reflected in the ingset status even though it's async.